internal/pkg/aws/ec2/ec2.go

```diff
@@ -377,3 +364,76 @@ func toEC2Filter(filters []Filter) []*ec2.Filter {
 	}
 	return ec2Filter
 }
+
+type routeTable ec2.RouteTable
+
+// IsMain returns true if the route table is the default route table for the VPC.
+// If a subnet is not associated with a particular route table, then it will default to the main route table.
+func (rt *routeTable) IsMain() bool {
+	for _, association := range rt.Associations {
+		if aws.BoolValue(association.Main) {
+			return true
+		}
+	}
+	return false
+}
+
+// HasIGW returns true if the route table has a route to an internet gateway.
+func (rt *routeTable) HasIGW() bool {
+	for _, route := range rt.Routes {
+		if strings.HasPrefix(aws.StringValue(route.GatewayId), internetGatewayIDPrefix) {
+			return true
+		}
+	}
+	return false
+}
+
+// AssociatedSubnets returns the list of subnet IDs associated with the route table.
+func (rt *routeTable) AssociatedSubnets() []string {
```

Comment:
Does this function have to be public? same for the other new added public functions.

Reply:
The type itself is private so I don't think it particularly matters since it doesn't get exposed by the package 🤷 do you know if there is a particular benefit? 
I marked it as public to indicate that they can be used by clients, and i would mark it as private if only `*routeTable` should use it